crypto/x509/x_all.c

```diff
@@ -138,12 +175,58 @@ X509 *X509_load_http(const char *url, BIO *bio, BIO *rbio, int timeout)
                                    ASN1_ITEM_rptr(X509));
 }
 
+/*
+ * Drop empty SKID or AKID X509_REQ extensions that might result from:
+ * "...KeyIdentifier = none"
+ */
+
+static void drop_empty_req_keyid_exts(X509_REQ *x)
+{
+    STACK_OF(X509_EXTENSION) *exts;
+    int i, n;
+
+    if ((exts = ossl_x509_req_get1_extensions_by_nid(x, NID_ext_req)) == NULL
+        || (n = sk_X509_EXTENSION_num(exts)) < 1)
```

Comment:
This leaks `exts` in case it is an empty list.

BTW, I wonder why you made the implementation of the two drop_ function more complicated?